pkg/registries/registries_test.go

```diff
@@ -9,31 +9,62 @@ import (
 	"testing"
 
 	"github.com/BurntSushi/toml"
-	"github.com/containers/image/pkg/sysregistriesv2"
-	"github.com/containers/image/types"
+	"github.com/containers/image/v5/pkg/sysregistriesv2"
+	"github.com/containers/image/v5/types"
 	apioperatorsv1alpha1 "github.com/openshift/api/operator/v1alpha1"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"k8s.io/apimachinery/pkg/util/diff"
 )
 
-func TestScopeMatchesRegistry(t *testing.T) {
+func TestRegMatchesScope(t *testing.T) {
 	for _, tt := range []struct {
 		scope, reg string
 		expected   bool
 	}{
-		{"quay.io", "example.com", false},             // Host mismatch
-		{"quay.io", "quay.io", true},                  // Host match
-		{"quay.io:443", "quay.io", false},             // Port mismatch (although reg is a prefix of scope)
-		{"quay.io:443", "quay.io:444", false},         // Port mismatch
-		{"quay.io.example.com", "quay.io", false},     // Host mismatch (although reg is a prefix of scope)
-		{"quay.io2", "quay.io", false},                // Host mismatch (although reg is a prefix of scope)
-		{"quay.io/ns1", "quay.io", true},              // Valid namespace
-		{"quay.io/ns1/ns2/ns3", "quay.io", true},      // Valid namespace
-		{"quay.io/ns1/ns2/ns3", "not-quay.io", false}, // Host mismatch
+		{"quay.io", "example.com", false},                   // Host mismatch
+		{"quay.io", "quay.io", true},                        // Host match
+		{"quay.io:443", "quay.io", false},                   // Port mismatch (although reg is a prefix of scope)
+		{"quay.io:443", "quay.io:444", false},               // Port mismatch
+		{"quay.io.example.com", "quay.io", false},           // Host mismatch (although reg is a prefix of scope)
+		{"quay.io2", "quay.io", false},                      // Host mismatch (although reg is a prefix of scope)
+		{"quay.io/ns1", "quay.io", true},                    // Valid namespace
+		{"quay.io/ns1/ns2/ns3", "quay.io", true},            // Valid namespace
+		{"quay.io/ns1/ns2/ns3", "not-quay.io", false},       // Host mismatch
+		{"bar/example.foo", "*.foo", false},                 // Invalid wildcard entry
+		{"example/bar.foo/quay.io", "*.foo", false},         // Invalid wildcard entry
+		{"example/bar.foo:400", "*.foo", false},             // Invalid wildcard entry
+		{"foo.example.com", "*.example.com", true},          // Valid sub-scope match
+		{"*.foo.example.com", "*.example.com", true},        // Valid sub-scope match
+		{"foo.example.com/bar", "*.example.com", true},      // Valid sub-scope match
+		{"foo.registry.com", "*.example.com", false},        // Invalid sub-scope match
+		{"foo.example.com", "**.example.com", false},        // Invalid sub-scope entry
+		{"foo.example.com", "example.*.com", false},         // Invalid sub-scope entry
+		{"foo.example.com", "*.example.com/foo/bar", false}, // Invalid wildcard entry
```

Comment:
could you also include `{"foo.example.com:443/bar/baz", "*.example.com", true}` and `{"foo.example.com:443/bar/bz", "*.example.com/bar/baz", false}` to check for namespace/repo after port number?

Reply:
Added, thanks!